.github/workflows/continous-integration.yml

```diff
@@ -276,7 +276,107 @@ jobs:
         JOBS: 2
         PYTHONIOENCODING: "utf-8"
       run: |
-        make test
+        make test-non-training
+
+    - name: Send Coverage Report 📊
+      if: needs.changes.outputs.backend == 'true' && matrix.python-version == 3.6 && matrix.os != 'windows-latest'
+      env:
+        COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        COVERALLS_SERVICE_NAME: github
+      run: poetry run coveralls
+
+  test_training:
```

Comment:
Is there an easy way to reuse these duplicate steps?

Reply:
No, you can consider using GH action of the composite type but I wouldn't say it's an easy way

https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action